### PR TITLE
Stencil Outlines

### PR DESCRIPTION
**Pull request by Cameron-Micka**
## Overview
This PR adds features to the MeshOutline component so that it can work with scenes with skyboxes or other environmental features (respects depth values from objects in the environment). To enable this feature, check the "Use Stencil Outline" option:
image

This feature works via the following process:

1. One pass for the normal object (this allows people to use any shader/material setup they want)
2. A second pass that writes to the stencil buffer (this can also be configured to use the material from the 1st pass and save a pass)
3. Finally, a pass that renders the outline where the stencil test passes

You can also specify a margin to produce a gap between the object and outline:
image

Constant width outlines (same width no matter how far from the camera) are also possible via this material check box:
image

It requires a lot of careful setup to get this process working. Because of that I've adopted and modified the "setup wizard" from MRTKv2. The wizard will help you automatically create and modify materials at edit time:
image

The sample gallery has also been updated to demo these features:
Screenshot_2022-11-17_10-39-07-AM_bf10c17b62a8b7641b57e67caae09459

Things this PR doesn't solve:

- Outline materials are shared, so the user will need to be careful when sharing materials in a scene. For example if you are outlining a left and right hand, you will want separate materials for the left and right hand. 
- Overlapping outlines will show stencil artifacts if a large margin is used. This can be mitigated my changing stencil references and masks. This is not something automated by the component since many projects have bespoke stencil setups.  

## Changes
- Fixes: #8 and #33

### CODE REVIEW

**Comment by Cameron-Micka** [comment]
Gif of the outline when animated:
Anim

File: com.microsoft.mrtk.graphicstools.unity/Editor/Utilities/ScreenshotUtilities.cs [hunk 1]
```diff
@@ -101,24 +101,15 @@ public static bool CaptureScreenshot(string path, int superSize = 1, bool transp
                 }
             }
 
-            // Create a camera clone with a transparent clear color.
+            // Create a camera clone.
             var renderCamera = new GameObject().AddComponent<Camera>();
+            renderCamera.CopyFrom(camera);
```
Comment (Cameron-Micka) [inline com.microsoft.mrtk.graphicstools.unity/Editor/Utilities/ScreenshotUtilities.cs:106]: Unrelated to outlines but had this tiny fix already in the works. Ensures screenshots have an opaque background when requested.

**Comment by Species521** [comment]
Looks amazing, almost perfect, just found two tiny inconsistencies.
When the object is really complex/self-occluding, small artefacts are visible upon geometry overlaps:
Screenshot 2022-11-09 144745

If this is just too rare and proves negligible, a workaround might be to just increase Outline Margin:
Screenshot 2022-11-09 144805

A visual issue can occur if a second object with the same stencil reference is present inside/behind the margin, where it falls victim to the stencil of the first object in the foreground, here exaggerated on purpose:
Screenshot 2022-11-09 144659
No fix for that of course, but needs pointing out in the doc.

Great work really.

**Comment by Cameron-Micka** [comment]
> Looks amazing, almost perfect, just found two tiny inconsistencies.

Thank you @Species521! I've made a few small changes that hopefully address these issues: 
image

**Pull request merged**